docs/troubleshooting/detections/detection-rules.asciidoc

```diff
@@ -50,6 +50,14 @@ This can be caused by bad indicator data, a misconfigured rule, or too many even
 If you receive the following rule failure: `"An error occurred during rule execution: message: "Request Timeout after 90000ms"`, this indicates that the query phase is timing out. Try refining the time frame or dividing the data defined in the query into multiple rules.
 ====
 
+[discrete]
+[[IM-rule-heap-memory]]
+.Indicator match rules are failing due to insufficient heap memory
+[%collapsible]
+====
+If you receive the following rule failure: `Bulk Indexing of signals failed: index: ".index-name" reason: "maxClauseCount is set to 1024" type: "too_many_clauses"`, this indicates that the limit for the total number of clauses that a query tree can have is too low. Try increasing your cluster's heap memory to 2GB or more.
```

Comment:
@nkhristinin thanks for all the feedback!

I can revise the tip for increasing the ES JVM heap size and then point to the docs for doing that. Just want to verify that we're recommending that users set their heap size to 1GB and not 2GB (this was the amount I had taken down in my notes after researching the related dev issue and us meeting last week).

For the troubleshooting tip title, how about:
Indicator match rules are failing with a `maxClauseCount` error.

The version above doesn't describe the reason for the rules failing, but does indicate that the max clause count is involved. If 

Another revision, which is a bit more descriptive of the underlying cause: 
Indicator match rules are failing because the `maxClauseCount` limit is too low.

Let me know what you think - thanks!